src/errors.rs

```diff
@@ -200,9 +216,13 @@ impl ErrorCode {
     }
 }
 
-impl From<c_abi::xdrfile::BindgenTy1> for ErrorCode {
-    fn from(code: c_abi::xdrfile::BindgenTy1) -> Self {
-        match code {
+impl<T> From<T> for ErrorCode
+where
+    T: TryInto<c_abi::xdrfile::BindgenTy1>,
+    <T as TryInto<c_abi::xdrfile::BindgenTy1>>::Error: std::fmt::Debug,
+{
+    fn from(code: T) -> Self {
+        match code.try_into().expect("C API return code was out of range") {
```

Comment:
The issue is probably the autogenerated code from bindgen for the c part. Error codes (BindgenTy1) are `u32` right now, but all methods return `i32` (`c_int`). This doesn't really make any sense. Since all possible error codes are in the range of 0-13, we can probably just make BindgenTy1 an i32 and remove this cast.

Reply:
Done! I had this thought too but I didn't really wanna go messing around with the bindgen stuff. But it turned out to be easy!